### PR TITLE
Internsystem with frontend using Docker on new server

### PR DESCRIPTION
**Pull request by henrist**
There is also a change on url, from `internt.cyb.no` to just `in.cyb.no`. Access to UiO weblogin must be provided to this new address before it can go live.

On deployments there will be made backups of internsystem that is uploaded to UiO. (Today this is stored locally on deployment.)

`test` branch is used for https://test.in.cyb.no/ (which uses `test`-backend) and https://frontend.test.in.cyb.no/ (which uses `prod`-backend)

### CODE REVIEW

File: internsystem/shell.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# be removed after use.
+
+command=sh
+if [ -n "$1" ]; then
+    command="$@"
+fi   
+
+. load-vars.sh
+
+SECRETKEY=$(<$keyfile)
```
Comment (henrist) [inline internsystem/shell.sh]: these two lines are set in load-vars.sh

File: internsystem/run.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+. load-vars.sh
+
+running=$(docker inspect --format="{{ .State.Running }}" cyb-internsystem 2>/dev/null)
```
Comment (henrist) [inline internsystem/run.sh]: This wont work with `set -e` on first run

File: internsystem/update.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+set -e
+
+./update-source.sh
+./build.sh
+
+if [ "$1" != "init" ]; then
+    ./backup.sh
+fi
+
+running=$(docker inspect --format="{{ .State.Running }}" cyb-internsystem 2>/dev/null)
```
Comment (henrist) [inline internsystem/update.sh]: This wont work with `set -e` on first run

File: internsystem-frontend/build.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e
+
+if ! [ -d src ]; then
+    echo "No source files exists, please run ./update-source.sh"
+    exit 1
+fi
+
+commit=$(cd src && git rev-parse --short HEAD)
+docker build -t cyb/internsystem-frontend:$commit .
```
Comment (henrist) [inline internsystem-frontend/build.sh]: also tag as latest

File: internsystem/build.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e
+
+if ! [ -d src ]; then
+    echo "No source files exists, please run ./update-source.sh"
+    exit 1
+fi
+
+commit=$(cd src && git rev-parse --short HEAD)
+docker build -t cyb/internsystem:$commit .
```
Comment (henrist) [inline internsystem/build.sh]: also tag as latest

File: internsystem/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,34 @@
+FROM python:3-alpine
+
+# postgresql-dev required to build psycopg2 with pip
+RUN apk --update add \
+      ca-certificates \
+      build-base \
+      libxml2-dev \
+      libxslt-dev \
+      postgresql-dev \
+      py-gunicorn
```
Comment (henrist) [inline internsystem/Dockerfile]: Don't need this? The one from pip is used?

File: internsystem-frontend/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FROM node:6
+
```
Comment (henrist) [inline internsystem-frontend/Dockerfile:3]: Add maintainer (see also other Dockerfiles)

File: internsystem/shell.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# This script will set up a temporary container and optionally run
+# the provided command, or else start a shell. The container will
+# be removed after use.
+
+command=sh
+if [ -n "$1" ]; then
+    command="$@"
+fi
```
Comment (henrist) [inline internsystem/shell.sh]: is it extra whitespace at end of line?

File: internsystem-backend/environments/prod/settings_local.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+#]
+
+# if collectstatic is used, set up this
+STATIC_ROOT = "/usr/src/static/"
+
+# example for postgres configuration
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': os.environ['POSTGRES_NAME'],
```
Comment (matsasc) [inline internsystem-backend/environments/prod/settings_local.py:28]: Where do you set these environment variables?

**Comment by matsasc** [comment]
Looks good, would prefer if you had used the same names in the setting files as in the vars files. Makes it a bit easier to see what you set where.

File: internsystem-backend/environments/prod/settings_local.py (hunk 9 above)
Comment (henrist) [inline internsystem-backend/environments/prod/settings_local.py:28]: With Docker it is normal to inject such environment variables when the container is run, see `run.sh`.

**Comment by henrist** [comment]
> Looks good, would prefer if you had used the same names in the setting files as in the vars files. Makes it a bit easier to see what you set where.

A good reason for not doing this is that the variables in the `.vars.sh`-file is not available in the settings-file, so doing such might cause more confusion.

**Comment by matsasc** [comment]
Why didn't you use the env-file option? ref: https://docs.docker.com/v1.11/engine/reference/commandline/run/

**Comment by henrist** [comment]
> Why didn't you use the env-file option? ref: https://docs.docker.com/v1.11/engine/reference/commandline/run/

Because not all options in the `vars.sh`-file is for the container application itself. Also it don't support the scripting that is required for the secret values that's read from other files. I think specifying this explicitly on the run-command is the most transparent way to know what is available inside the container.

**Pull request merged**